<img style="float: left; margin: 30px 15px 15px 15px;" src="https://pngimage.net/wp-content/uploads/2018/06/logo-iteso-png-5.png" width="300" height="500" /> 
    
    
### <font color='navy'> Simulación de procesos financieros. 

**Nombres:** Alan Jose Maldonado Garcia, Ana Luisa Espinoza López.

**Fecha:** 19 de Septiembre del 2023.

**Expediente** :732884,734753.
**Profesor:** Oscar David Jaramillo Zuluaga.
    
**Link Github**: https://github.com/AlanMaldonado1998/ProyectoConjunto_AJMaldonado_ALEspinoza

# Tarea 4: Clase 9

## Enunciado de tarea 


### Ejercicio 1
Como ejemplo simple de una simulación de Monte Carlo, considere calcular la probabilidad de una suma particular del lanzamiento de tres dados (cada dado tiene valores del uno al seis). Además cada dado tiene las siguientes carácterísticas: el primer dado no está cargado (distribución uniforme todos son equiprobables); el segundo y tercer dado están cargados basados en una distribución binomial con parámetros (`n=5, p=0.5` y `n=5, p=0.2`). Calcule la probabilidad de que la suma resultante sea 7, 14 o 18. 

### Ejercicio 1:

Código de solución Alan

In [ ]:
# Código de solución estudiante 1
.
.
.
.


Código de solución Ana Luisa

In [1]:
import numpy as np
import scipy.stats as st

In [2]:
def prob_dados(s,N):  #s:suma resultante deseada   N:número de tiros
    dado_1 = [np.random.randint(1,7) for i in range(0,N)]
    dado_2 = (st.binom(n=5, p=0.5).rvs(N))+1
    dado_3 = (st.binom(n=5, p=0.2).rvs(N))+1
    suma = [(dado_1[i]+dado_2[i]+dado_3[i]) for i in range(N)]
    c = suma.count(s)
    p = round(((c/N)*100),2)
    msj = f"La probabilidad de que la suma sea {s} es de {p}%"
    return msj

In [3]:
print(prob_dados(7,1000))
print(prob_dados(14,1000))
print(prob_dados(18,1000))

La probabilidad de que la suma sea 7 es de 12.1%
La probabilidad de que la suma sea 14 es de 1.5%
La probabilidad de que la suma sea 18 es de 0.0%


# 2 Ejercicio de aplicación- Cafetería Central

Premisas para la simulación:
- Negocio de alimentos que vende bebidas y alimentos.
- Negocio dentro del ITESO.
- Negocio en cafetería central.
- Tipo de clientes (hombres y mujeres).
- Rentabilidad del 60%.

## Objetivo
Realizar una simulación estimado el tiempo medio que se tardaran los clientes en ser atendidos entre el horario de 6:30 a 1 pm. Además saber el consumo. 
**Analizar supuestos y limitantes**

## Supuestos en simulación
Clasificación de clientes: 
- Mujer  = 1 $\longrightarrow$ aleatorio < 0.5
- Hombre = 0 $\longrightarrow$ aleatorio $\geq$ 0.5.

Condiciones iniciales:
- Todas las distrubuciones de probabilidad se supondrán uniformes.
- Tiempo de simulación: 6:30 am - 1:30pm $\longrightarrow$ T = 7 horas = 25200 seg.
- Tiempo de llegada hasta ser atendido: Min=5seg, Max=30seg.
- Tiempo que tardan los clientes en ser atendidos:
    - Mujer: Min = 1 min= 60seg, Max = 5 min = 300 seg
    - Hombre: Min = 40 seg, Max = 2 min= 120 seg
- Consumo según el tipo de cliente:
    - Mujer: Min = 30 pesos, Max = 100 pesos
    - Hombre: Min = 20 pesos, Max = 80 pesos

Responder las siguientes preguntas basados en los datos del problema:
1. ¿Cuáles fueron los gastos de los hombres y las mujeres en 5 días de trabajo?.
2. ¿Cuál fue el consumo promedio de los hombres y mujeres?
3. ¿Cuál fue el número de personas atendidas por día?
4. ¿Cuál fue el tiempo de atención promedio?
5. ¿Cuánto fue la ganancia promedio de la cafetería en 5 días de trabajo y su respectiva rentabilidad?

### Ejercicio 2: 

Código de solución Alan

In [ ]:
# Código de solución estudiante 1
.
.
.
.


Código de solución Ana Luisa

In [4]:
import numpy as np
import random

In [5]:
################## Datos del problema
d = 5 #dias de trabajo
T =25200 #tiempo total simulación
#tiempo de llegada 
T_at_min = 5; T_at_max = 30
#tiempo en que son atendidos
T_mujer_min =60; T_mujer_max = 300
T_hombre_min = 40; T_hombre_max = 120
C_mujer_min = 30; C_mujer_max = 100 #datos de consumo mujeres
C_hombre_min = 20; C_hombre_max = 80 #datos de consumo hombres

### Variables adicionales
ct_m = np.array([])
ct_h = np.array([])
t_at = np.array([])
cl_dia = [[] for _ in range(d)]
g_dia = [[] for _ in range(d)]

In [6]:
#### funciones para determinar tipo de cliente, su consumo y el tiempo de atención de cada uno

# función que me da el tiempo total (atención + llegada) y consumo de un hombre y de una mujer y el tiempo de atención
def hombre():
    at = np.random.uniform(T_hombre_min, T_hombre_max)
    tt = at + np.random.uniform(T_at_min, T_at_max)
    c = np.random.uniform(C_hombre_min, C_hombre_max)

    return [tt,c,at]

def mujer():
    at = np.random.uniform(T_mujer_min, T_mujer_max)
    tt = at + np.random.uniform(T_at_min, T_at_max)
    c = np.random.uniform(C_mujer_min, C_mujer_max)

    return [tt,c,at]

# función para determinar el tipo de cliente 1 si es mujer, 0 si es hombre
def tipo():
    j=np.random.uniform(0,1)
    if j<0.5:
        return 1
    return 0

In [7]:
for i in range(d):
    c_dia = 0 #clientes del día
    t_trans = 0 #tiempo transcurrido
    g = 0 #ganancia por día
    while t_trans<T:
        cliente = tipo()
        if cliente == 1:
            tiempo,cons,a = mujer()
            ct_m = np.append(ct_m,cons)
        else:
            tiempo,cons,a = hombre()
            ct_h = np.append(ct_h,cons)
        t_trans += tiempo
        g += cons
        t_at = np.append(t_at,a)
        c_dia += 1
    cl_dia[i]=c_dia
    g_dia[i]=g

# ganancias totales 
ganancia_t = np.array(g_dia).sum()

In [9]:
print(f"1. En 5 días, los gastos de hombres fueron por: ${round(ct_h.sum(),2)} y de mujeres por: ${round(ct_m.sum(),2)}")
print(f"2. El consumo promedio de hombres fue de: ${round(ct_h.mean(),2)} y de mujeres: ${round(ct_m.mean(),2)}")
print(f"3. El número de personas atendias por día es de {cl_dia}, respectivamente")
print(f"4. El tiempo de atención promedio fue de  {round(t_at.mean(),2)} segundos")
print(f"5. La ganancia promedio en 5 días es de ${round(ganancia_t/d,2)} \n y la rentabilidad de: ${round((ganancia_t*0.6)/d,2)}")

1. En 5 días, los gastos de hombres fueron por: $18900.14 y de mujeres por: $28955.8
2. El consumo promedio de hombres fue de: $49.74 y de mujeres: $64.35
3. El número de personas atendias por día es de [157, 171, 158, 167, 177], respectivamente
4. El tiempo de atención promedio fue de  135.26 segundos
5. La ganancia promedio en 5 días es de $9571.19 
 y la rentabilidad de: $5742.71


### Respuesta planteada en el ejercicio usando (markdown)
1. ¿Cuáles fueron los gastos de los hombres y las mujeres en 5 días de trabajo?.
    - gastos de hombres: 18,900.14
    - gastos de mujeres: 28,955.8
    
    
2. ¿Cuál fue el consumo promedio de los hombres y mujeres?
    - gasto promedio de hombres: 49.74
    - gasto promedio de mujeres: 64.35
    
    
3. ¿Cuál fue el número de personas atendidas por día?
    - 157, 171, 158, 167 y 177, respectivamente
    
    
4. ¿Cuál fue el tiempo de atención promedio?
    - El tiempo de atención promedio fue de 135.26 segundos por cliente
    
    
5. ¿Cuánto fue la ganancia promedio de la cafetería en 5 días de trabajo y su respectiva rentabilidad?
    - la ganancia promedio en 5 días fue de : 9,571.19
    - la rentabilidad promedio de: 5,742.71 